torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1445,6 +1445,8 @@ class DimDynamic(Enum):
     SIZE_LIKE_UNBACKED = 3
```

Comment:
I find this naming super confusing. Let me get this straight:

1) `torch._dynamo.mark_unbacked` sets the dimension to DimDynamic.SIZE_LIKE_UNBACKED via https://www.internalfb.com/code/fbsource/[7157ca20df68]/fbcode/caffe2/torch/_dynamo/variables/builder.py?lines=2738
2) In https://github.com/pytorch/pytorch/issues/137100, you write this will "opt into the alternate size oblivious semantics, which allows you to avoid doing specializations on 0/1 in many situations"

So it's really not SIZE_LIKE_UNBACKED, but really more like SIZE_OBLIVIOUS_UNBACKED no?

Reply:
Uhm, I think I might be confusing myself. So it seems like size_oblivious and size_like are actually indeed analogous concepts. 

So size_oblivious seems to describe a type of analysis where we assume all size-like symbols are >=2 for purpose of analysis.

Size-like, is just an annotation that we put onto symbols that size_oblivious uses to narrow down the symbols with which we can do size oblivious analysis.

So I guess now my confusion is why we have an OBLIVIOUS_SIZE. Why not call it SIZE_LIKE_BACKED?

Reply:
Ah.. so I guess what separates oblivious and size-like is the counter-factual reasoning you do. So SIZE_LIKE_UNBACKED does size oblivious reasoning BUT will raise when we try to guard. OBLIVIOUS_SIZE also does size oblivious reasoning BUT will only raise if the counter-factual reasoning of non-size-like vs size-like differs.

Reply:
I wonder if it makes more sense to do something like OBLIVIOUS_SIZE_LIKE_UNBACKED so future readers don't need to wander through this idea maze.

Reply:
`SIZE_LIKE_BACKED` might be OK, but it might be confusing because we do in fact allocate unbacked symbols for `OBLIVIOUS_SIZE` (and they just get special handling due to the `oblivious_var_to_val` map). `OBLIVIOUS_SIZE_LIKE_UNBACKED` could potentially make what is currently called `SIZE_LIKE_UNBACKED` more clear, although for more symmetry I would probably call it `OBLIVIOUS_UNBACKED_SIZE` if we weren't renaming. This is all private API so it's easy to change, just need to decide what is clear.